plugins/readtimer.koplugin/main.lua

```diff
@@ -365,4 +303,110 @@ function ReadTimer:onResume()
     end
 end
 
+function ReadTimer:onShowAlarm(touchmenu_instance)
+    local now_t = os.date("*t")
+    local curr_hour = now_t.hour
+    local curr_min = now_t.min
+    local time_widget = DateTimeWidget:new{
+        hour = curr_hour,
+        min = curr_min,
+        ok_text = _("Set alarm"),
+        title_text =  _("New alarm"),
+        info_text = _("Enter a time in hours and minutes."),
+        callback = function(alarm_time)
+            self:SetAlarm(alarm_time, now_t, touchmenu_instance)
+        end
+    }
+    self:addCheckboxes(time_widget)
+    UIManager:show(time_widget)
+
+end
+
+function ReadTimer:SetAlarm(alarm_time, then_t, touchmenu_instance)
+    if not then_t then
```

Comment:
How can this happen?

Reply:
See [above](https://github.com/koreader/koreader/pull/14061#discussion_r2209659137), the concept behind these functions in my quick concept patch was to call onSetAlarm and onSetInterval without displaying a dialog.[1] The way this has been rewritten of course it doesn't make sense anymore, and it can be readded if someone decides to implement that after all.

[1] Concretely, you'd set up an action to set the alarm at 23:30 for example.

Reply:
Pardon, I was a bit unclear but that's not why I quickly wrote that patch a few months ago and forgot about it until I saw that issue. ;-)